src/core/orderbookutils.js

```diff
@@ -71,7 +109,16 @@ OrderBookUtils.getOfferQuality = function(offer, currencyGets) {
       reference_date: new Date()
     });
   } else {
-    amount = Amount.from_json(offer.quality + IOU_SUFFIX);
+
+    let currency = offer.TakerGets.currency;
+    let issuer = offer.TakerGets.issuer;
+
+    if (currency === undefined || issuer === undefined) {
+      currency = offer.TakerPays.currency;
```

Comment:
As always using IOU_SUFFIX before suggests, the issuer/currency don't matter at all. All this seems redundant when createAmount will set some neutral values anyway.